docs/custom-code/prisma-custom-attributes.md

```diff
@@ -0,0 +1,326 @@
+---
+id: prisma-custom-attributes
+title: Customize your Entities and Fields with Custom Prisma Attributes
+sidebar_label: Customize your Entities and Fields with Custom Prisma Attributes
+slug: /custom-code/custom-prisma-attributes
+---
+
+# Customizing your Entities and Fields with Custom Prisma Attributes
+
+Prisma is a central cog in [Amplication's open-source stack](/getting-started/) for creating and managing your generated app's database.
+
+Prisma's ability to automatically generate fully customizable database schema helps to keep you focused on building new app features rather than hassling with refactoring for entity changes.
+
+Amplication's UI allows you to declare entities and fields in an intuitive way that map back to common Prisma database schema.
+But, what if you need to further customize your entities?
+What if you need additional functionality on your entity's fields?
+
+The best way to do this is with Prisma's _Custom Attributes_.
+
+In this guide, you'll learn how to fully utilize Prisma's _Custom Attributes feature_ through the **Custom Attributes text field** available in your project's _Entities_ and _Fields_ settings page.
+
+Let's get started.
+
+## How Amplication generates your database schema with Prisma
+
+First, a quick primer on how Amplication manages and creates your app's database.
+
+Every action that you take in Amplication's UI related to database changes is mapped back to the code that Amplication creates for your project.
+
+For example, let's say you create a new service and choose MySQL as your database of choice.
+Amplication will generate the following Prisma schema:
+
+```
+datasource mysql {
+  provider = "mysql"
+  url      = env("DB_URL")
+}
+
+generator client {
+  provider = "prisma-client-js"
+}
+
+model User {
+  createdAt DateTime @default(now())
+  firstName String?
+  id        String   @id @default(cuid())
+  lastName  String?
+  password  String
+  roles     Json
+  updatedAt DateTime @updatedAt
+  username  String   @unique
+}
+```
+
+:::important
+The _User_ entity is always created by default for every Amplication service.
```

Comment:
@dericksozo this is no longer accurate. following recent changes the User entity is automatically created if a service uses authentication but for a service created without Auth there will not be a User entity.